### PR TITLE
feat(inputs.exec): Do not truncate stderr output in debug mode

### LINKED ISSUE #15784: feat: allow input.exec plugins to enable the debug setting
### Use Case

I am trying to debug an exec command run that crashes with a stack trace.
I can't see the contents of the stack trace due to the truncation performed by Exec.

Exec has a debug flag internally that would allow me to skip truncation and see the full stack trace, but there's no way to set that option in the configuration file.

### Expected behavior

Exec should support a `debug` field in its config that would set the plugin's internal `debug` flag.

### Actual behavior

There's no option to set the debug flag.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by ltadeut**
## Summary

This PR adds a debug option to the exec plugin's toml config.
The option is used to set the internal `CommandRunner.debug`, which in turn disables stderr truncation.

This is necessary in order read stderr messages while debugging exec plugin issues.

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

resolves #15784

### CODE REVIEW

**Comment by ltadeut** [comment]
!signed-cla

File: plugins/inputs/exec/dev/null
Comment (DStrand1) [inline plugins/inputs/exec/dev/null:1]: Is this file needed?

File: plugins/inputs/exec/exec.go [hunk 1]
```diff
@@ -49,19 +50,24 @@ type Exec struct {
 
 func NewExec() *Exec {
 	return &Exec{
-		runner:  CommandRunner{},
+		runner:  &CommandRunner{},
```
Comment (DStrand1) [inline plugins/inputs/exec/exec.go:53]: Since this needs to change to a pointer, can you update other methods to use a pointer? Ex: the truncate() method and any others

File: plugins/inputs/exec/sample.conf [hunk 2]
```diff
@@ -28,3 +28,7 @@
   ## them here:
   ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_INPUT.md
   # data_format = "json"
+
+  ## Run the exec plugin in debug mode
+  ## If set to true, exec will not truncate the command's stderr output.
+  # debug = false
```
Comment (DStrand1) [inline plugins/inputs/exec/sample.conf:34]: Could you also add this to the sample config in the readme?

**Review by DStrand1 — COMMENTED** [review]
Thanks for the PR! Could you address linter issues, as well as my handful of comments? Thanks!

File: plugins/inputs/exec/dev/null
Comment (ltadeut) [reply plugins/inputs/exec/dev/null:1]: Bad checking, removed!

File: plugins/inputs/exec/exec.go (hunk 1 above)
Comment (ltadeut) [reply plugins/inputs/exec/exec.go:53]: all done!

File: plugins/inputs/exec/sample.conf (hunk 2 above)
Comment (ltadeut) [reply plugins/inputs/exec/sample.conf:34]: Done! Did that by running `make docs`

**Review by DStrand1 — APPROVED** [review]
Looks great, thanks for wrapping this up!

File: plugins/inputs/exec/exec.go [hunk 3]
```diff
@@ -49,20 +50,25 @@ type Exec struct {
...
 }
 
 type CommandRunner struct {
 	debug bool
 }
 
-func (c CommandRunner) truncate(buf bytes.Buffer) bytes.Buffer {
+func (c *CommandRunner) SetDebug(debug bool) {
+	c.debug = debug
+}
```
Comment (srebhan) [inline plugins/inputs/exec/exec.go:69]: Instead of using this function you could directly set the variable, can't you?

File: plugins/inputs/exec/exec.go [hunk 4]
```diff
@@ -149,6 +155,9 @@ func (e *Exec) SetParser(parser telegraf.Parser) {
 
 func (e *Exec) Gather(acc telegraf.Accumulator) error {
 	var wg sync.WaitGroup
+
+	e.runner.SetDebug(e.Debug)
```
Comment (srebhan) [inline plugins/inputs/exec/exec.go:159]: Please do this in `Init()` **once** instead of setting the same value in each gather cycle.

File: plugins/inputs/exec/sample.conf (hunk 2 above)
Comment (srebhan) [inline plugins/inputs/exec/sample.conf:34]: Please do not introduce a new variable! Instead reuse the log-level of the `telegraf.Logger` which can be accessed by `Log.Level()`.

**Review by srebhan — COMMENTED** [review]
@ltadeut thanks for your contribution! Some small comments from my side...

File: plugins/inputs/exec/exec.go (hunk 4 above)
Comment (ltadeut) [reply plugins/inputs/exec/exec.go]: Done!

File: plugins/inputs/exec/sample.conf (hunk 2 above)
Comment (ltadeut) [reply plugins/inputs/exec/sample.conf]: Done. Changed the code to get the log level when Init() runs and set `debug` to true when Trace is set.

File: plugins/inputs/exec/exec.go (hunk 3 above)
Comment (srebhan) [reply plugins/inputs/exec/exec.go]: Why not just initialize the runner in `Init()`?

Comment (ltadeut) [reply plugins/inputs/exec/exec.go]: Sounds good to me, will update the pr

**Review by Hipska — APPROVED** [review]
